templates/nuxt/store/crud.js

```diff
@@ -0,0 +1,273 @@
+      },
+      resetUpdate: ({ commit }) => {
+        commit(ACTIONS.RESET_UPDATE);
+      },
+      update: ({ commit }, item) => {
+        commit(ACTIONS.SET_ERROR, '');
+        commit(ACTIONS.TOGGLE_LOADING);
+
+        return service
+          .update(item)
+          .then(response => response.json())
+          .then(data => {
+            commit(ACTIONS.TOGGLE_LOADING);
+            commit(ACTIONS.SET_UPDATED, data);
+          })
+          .catch(e => handleError(commit, e));
+      }
+    },
+    getters: {
+      find: state => id => {
+        return resolveRelations(state.byId[id]);
+      },
+      getField,
+      list: (state, getters) => {
+        return state.allIds.map(id => getters.find(id));
+      }
+    },
+    mutations: {
+      updateField,
+      [ACTIONS.ADD]: (state, item) => {
+        Vue.set(state.byId, item['@id'], item);
+        Vue.set(state, 'isLoading', false);
+        if (state.allIds.includes(item['@id'])) return;
+        state.allIds.push(item['@id']);
+      },
+      [ACTIONS.RESET_CREATE]: state => {
+        Object.assign(state, {
+          isLoading: false,
+          error: '',
+          created: null,
+          violations: null
+        });
+      },
+      [ACTIONS.RESET_DELETE]: state => {
+        Object.assign(state, {
+          isLoading: false,
+          error: '',
+          deleted: null
+        });
+      },
+      [ACTIONS.RESET_LIST]: state => {
+        Object.assign(state, {
+          allIds: [],
+          byId: {},
+          error: '',
+          isLoading: false,
+          resetList: false
+        });
+      },
+      [ACTIONS.RESET_SHOW]: state => {
+        Object.assign(state, {
+          error: '',
+          isLoading: false
+        });
+      },
+      [ACTIONS.RESET_UPDATE]: state => {
+        Object.assign(state, {
+          error: '',
+          isLoading: false,
+          updated: null,
+          violations: null
+        });
+      },
+      [ACTIONS.SET_CREATED]: (state, created) => {
+        Object.assign(state, { created });
+      },
+      [ACTIONS.SET_DELETED]: (state, deleted) => {
+        if (!state.allIds.includes(deleted['@id'])) return;
+        Object.assign(state, {
+          allIds: remove(state.allIds, item => item['@id'] === deleted['@id']),
+          byId: remove(state.byId, id => id === deleted['@id']),
+          deleted
+        });
+      },
+      [ACTIONS.SET_ERROR]: (state, error) => {
+        Object.assign(state, { error, isLoading: false });
+      },
+      [ACTIONS.SET_SELECT_ITEMS]: (state, selectItems) => {
+        Object.assign(state, {
+          error: '',
+          isLoading: false,
+          selectItems
+        });
+      },
+      [ACTIONS.SET_TOTAL_ITEMS]: (state, totalItems) => {
+        Object.assign(state, { totalItems });
+      },
+      [ACTIONS.SET_UPDATED]: (state, updated) => {
+        Object.assign(state, {
+          byId: {
+            [updated['@id']]: updated
```

Comment:
I'm wondering whether it is intended to replace all existing items in `byId` after updating one item?